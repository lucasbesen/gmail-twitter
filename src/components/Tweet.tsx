import React from 'react';
import styled from 'styled-components';
import moment from 'moment';

import { ITweet } from '../types';

const Wrapper = styled.div`
  font-size: 14px;
  border-bottom: 1px solid #eee;
  padding: 14px;
  display: flex;
  cursor: pointer;

  :hover {
    box-shadow: 0px 1px 2px 0px #d3d3d3;
  }
`;

const TweetInfo = styled.div`
  display: flex;
  align-items: center;
  flex: 2;
  font-weight: 600;
`;

const TweetText = styled.div`
  flex: 8;
  padding-right: 15px;
`;

interface Props {
  tweet: ITweet;
}

const Tweet: React.FC<Props> = ({ tweet }) => (
  <Wrapper>
    <TweetInfo>{tweet.user.name}</TweetInfo>
    <TweetText>{tweet.text}</TweetText>
    <TweetInfo>{moment(new Date(tweet.created_at)).format('DD/MM/YYYY HH:mm')}</TweetInfo>
  </Wrapper>
);

export default Tweet;
