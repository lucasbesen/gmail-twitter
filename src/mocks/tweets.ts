import { ITweet } from '../types';

export const tweetMock: ITweet = {
  id: '11111111',
  text: 'This is a tweet',
  user: {
    name: 'User number 1',
  },
  created_at: 'Wed Aug 21 23:17:05 +0000 2019',
};
