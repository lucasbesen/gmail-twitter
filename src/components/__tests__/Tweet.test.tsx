import React from 'react';
import renderer from 'react-test-renderer';

import Tweet from '../Tweet';
import { tweetMock } from '../../mocks/tweets';

it('should render Tweet without error', () => {
  const component = renderer.create(<Tweet tweet={tweetMock} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
