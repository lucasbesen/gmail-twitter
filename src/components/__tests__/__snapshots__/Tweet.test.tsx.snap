// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Tweet without error 1`] = `
<div
  className="sc-iwsKbI jsRrR"
>
  <svg
    aria-hidden="true"
    className="sc-bwzfXH ldVAyv StyledIconBase-sc-bdy9j4 jbPeIR"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="sc-bZQynM htugFp StyledIconBase-sc-bdy9j4 jbPeIR"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
    />
  </svg>
  <div
    className="sc-gZMcBi NNoYB"
  >
    User number 1
  </div>
  <div
    className="sc-gqjmRU dFAKmp"
  >
    This is a tweet
  </div>
  <div
    className="sc-gZMcBi NNoYB"
  >
    21/08/2019 20:17
  </div>
</div>
`;
